### ISSUE TITLE
E0002 Unknown exception while processing rule E3020: expected string or bytes-like object

### ISSUE THREAD

**Issue by reversefold** [aws-cloudformation/cfn-python-lint#1310]
*cfn-lint version: (`cfn-lint 0.26.3`)*

*Description of issue.*

When running cfn-lint with a template that uses a macro the Route53 RecordSet rule E3020 assumed that A records will be strings.

For example:
```
  DomainNameReg:
    Type: AWS::Route53::RecordSet
    Properties:
      HostedZoneName:
        Fn::Sub: ${DnsEnvironmentDomain}.
      Name:
        Fn::Sub: ${DnsSubdomain}.${DnsEnvironmentDomain}
      Type: A
      TTL: '300'
      ResourceRecords:
        - Macro: {Ref: Param}
          Ref: MacroReplacedValue
```

The problem isn't that the template is flagged for violation of E3020, the problem is that the code fails and gives E0002.

When Adding `-D` the traceback I get is:
```
E0002 Unknown exception while processing rule E3020: Traceback (most recent call last):
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/site-packages/cfnlint/rules/__init__.py", line 203, in run_check
    return check(*args)
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/site-packages/cfnlint/rules/__init__.py", line 89, in wrapper
    results = match_function(self, filename, cfn, *args, **kwargs)
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/site-packages/cfnlint/rules/__init__.py", line 116, in matchall
    return self.match(cfn)  # pylint: disable=E1102
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/site-packages/cfnlint/rules/resources/route53/RecordSet.py", line 274, in match
    matches.extend(self.check_recordset(path, props, cfn))
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/site-packages/cfnlint/rules/resources/route53/RecordSet.py", line 201, in check_recordset
    check_value=self.check_a_record,
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/site-packages/cfnlint/__init__.py", line 613, in check_value
    value=value, path=new_path[:] + child_path, **kwargs))
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/site-packages/cfnlint/rules/resources/route53/RecordSet.py", line 52, in check_a_record
    if not re.match(REGEX_IPV4, value):
  File "/Users/jpatrin/.cache/pre-commit/repovfpiuzdp/py_env-python3.6/lib/python3.6/re.py", line 172, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or bytes-like object
```

Most of the other checks in the `RecordSet.py` seem to have an `if not isinstance(value, dict)` check, that would fix this problem as well.

Again, I'm not complaining about my non-standard template causing a valid error to be raised, that's expected and I can ignore those errors easily, as I have done in many other places.

**Comment by reversefold** [comment]
Thanks!
